extension/test/data_test.ts

```diff
@@ -69,4 +69,166 @@ describe('data.ts', function () {
+        kanji: '',
+        onkun: '',
+        nanori: '',
+        bushumei: '',
+        misc: {},
+        eigo: '',
+        hasNames: true,
+        data: [
+          { entry: '亜 [あ] /(f) A/', reason: undefined },
+          { entry: '阿 [あ] /(s) A/', reason: undefined },
+        ],
+        hasMore: false,
+        title: '',
+        index: 0,
+        matchLen: 1,
+      };
+
+      it('when max is 1, returns 1 data entry as text', function () {
+        const singleDataEntryAsText = rcxDict.makeText(
+          nameDictEntry,
+          /* max= */ 1
+        );
+
+        expect(singleDataEntryAsText).to.equal('亜\tあ\t(f) A\n');
+      });
+
+      it('when max is 2, returns 2 data entries as text', function () {
+        const twoDataEntriesAsText = rcxDict.makeText(
+          nameDictEntry,
+          /* max= */ 2
+        );
+
+        expect(twoDataEntriesAsText).to.equal('亜\tあ\t(f) A\n阿\tあ\t(s) A\n');
+      });
+
+      it('when max is greater than the number of data entries, returns all data entries as text', function () {
+        const allDataEntriesAsText = rcxDict.makeText(
+          nameDictEntry,
+          /* max= */ 1000
+        );
+
+        expect(allDataEntriesAsText).to.equal('亜\tあ\t(f) A\n阿\tあ\t(s) A\n');
+      });
+    });
+
+    describe('with a kanji dict entry', function () {
+      // kanjiDictEntry was generated with: rcxDict.kanjiSearch('両')
+      const kanjiDictEntry = {
+        kanji: '両',
+        onkun: 'リョウ、 てる、 ふたつ',
+        nanori: 'もろ',
+        bushumei: '',
+        misc: {
+          U: '4E21',
+          B: '1',
+          G: '3',
+          S: '6',
+          F: '247',
+          N: '34',
+          V: '23',
+          H: '3518',
+          DK: '2191',
+          DL: '2949',
+          L: '1168 both',
+          DN: '1252 both',
+          E: '411',
+          IN: '200',
+          P: '4-6-1',
+          I: '0a6.11',
+          Y: 'liang3',
+        },
+        eigo: 'both; old Japanese coin; counter for carriages (e.g., in a train); two',
+        hasNames: false,
+        data: [],
+        hasMore: false,
+        title: '',
+        index: 0,
+        matchLen: 0,
+      };
+
+      it('when max is 1, returns a single kanji data entry as text', function () {
+        const singleKanjiDataEntryAsText = rcxDict.makeText(
+          kanjiDictEntry,
+          /* max= */ 1
+        );
+
+        expect(singleKanjiDataEntryAsText).to.equal(
+          '両\nboth; old Japanese coin; counter for carriages (e.g., in a train); two\nリョウ、 てる、 ふたつ\n名乗り\tもろ\nHalpern\t3518\nHeisig 5th Edition\t1168 both\nHeisig 6th Edition\t1252 both\nHenshall\t411\nKanji Learners Dictionary\t2191\nKanji Learners Dictionary 2nd Edition\t2949\nNelson\t34\nNew Nelson\t23\nPinYin\tliang3\nSkip Pattern\t4-6-1\nTuttle Kanji & Kana\t200\nTuttle Kanji Dictionary\t0a6.11\nUnicode\t4E21\n'
+        );
+      });
+
+      it('when max is 1000, returns a single kanji data entry as text', function () {
```

Comment:
I would tweak this to make it more obvious that kanji entries ignore max essentially. Maybe:

`even with large max, returns a single kanji data entry as text`

Reply:
Yeah, that sounds better. I'll do that.